### PR TITLE
OCPBUGS-23947: set client options when interacting with azure api

### PR DESCRIPTION
**Pull request by flavianmissi**

### CODE REVIEW

File: pkg/asset/cluster/azure/azure.go [hunk 1]
```diff
@@ -50,8 +53,21 @@ func tagVNet(ctx context.Context, clusterID string, installConfig *installconfig
...
+				ActiveDirectoryAuthorityHost: session.Environment.ActiveDirectoryEndpoint,
+				Services: map[cloud.ServiceName]cloud.ServiceConfiguration{
+					cloud.ResourceManager: {
+						Audience: session.Environment.TokenAudience,
+						Endpoint: session.Environment.ResourceManagerEndpoint,
+					},
+				},
+			},
+		},
+	}
```
Comment (r4f4) [inline pkg/asset/cluster/azure/azure.go:68]: The session has the cloud config saved:
```suggestion
	clientOpts := &arm.ClientOptions{
		ClientOptions: azcore.ClientOptions{
			Cloud: session.CloudConfig,
		},
	}
```

File: pkg/destroy/azure/azure.go [hunk 2]
```diff
@@ -97,13 +99,27 @@ func (o *ClusterUninstaller) configureClients() error {
...
+					cloud.ResourceManager: {
+						Audience: o.Environment.TokenAudience,
+						Endpoint: o.Environment.ResourceManagerEndpoint,
+					},
+				},
+			},
+		},
+	}
+
+	rgClient, err := armresourcegraph.NewClient(o.tokenCreds, clientOpts)
```
Comment (r4f4) [inline pkg/destroy/azure/azure.go:116]:
```suggestion
	clientOpts := &arm.ClientOptions{
		ClientOptions: azcore.ClientOptions{
			Cloud:  session.CloudConfig,
		},
	}

	rgClient, err := armresourcegraph.NewClient(o.tokenCreds, clientOpts)
```

File: pkg/asset/cluster/azure/azure.go (hunk 1 above)
Comment (flavianmissi) [reply pkg/asset/cluster/azure/azure.go]: done, thanks

File: pkg/destroy/azure/azure.go (hunk 2 above)
Comment (flavianmissi) [reply pkg/destroy/azure/azure.go]: done.

File: pkg/destroy/azure/azure.go [hunk 3]
```diff
@@ -129,6 +138,7 @@ func New(logger logrus.FieldLogger, metadata *types.ClusterMetadata) (providers.
 	}
 
 	return &ClusterUninstaller{
+		Session:                     session,
```
Comment (flavianmissi) [inline pkg/destroy/azure/azure.go:141]: put it here for now so that the code works

**Review by r4f4 — COMMENTED** [review]
/lgtm

**Comment by r4f4** [comment]
`azure-ovn-shared-vpc`:
* install
```
time="2023-11-29T11:16:29Z" level=info msg="Creating infrastructure resources..."
time="2023-11-29T11:16:30Z" level=debug msg="Tagging vnet do-not-delete-shared-vnet-westus with kubernetes.io_cluster.ci-op-94k27pyp-51d74-dmlnl: shared"
```
* deprovision
```
time="2023-11-29T11:30:58Z" level=debug msg="removing shared tag from resource \"do-not-delete-shared-vnet-westus\"" resource=do-not-delete-shared-vnet-westus type=microsoft.network/virtualnetworks
time="2023-11-29T11:30:58Z" level=info msg="removed shared tag" resource=do-not-delete-shared-vnet-westus type=microsoft.network/virtualnetworks
```
`azurestack`:
* install
```
time="2023-11-29T11:26:27Z" level=info msg="Creating infrastructure resources..."
time="2023-11-29T11:26:27Z" level=debug msg="Tagging resource group ci-op-94k27pyp-195eb with kubernetes.io_cluster.ci-op-94k27pyp-195eb-9lgph: owned"
```

**Pull request merged**